src/js/mqtt.js

```diff
@@ -317,7 +323,7 @@ MqttClient.prototype.subscribe = function(topic, options, callback) {
   }
   try {
     var buf = this._handle._getSubscribe(topic, {
-      id: this._msgId++,
+      id: qosEqual0(options.qos) ? 0 : this._msgId++,
       qos: (options && options.qos) || 0,
```

Comment:
Subscribe's qos could be removed from this option, see https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901023, only the publish's qos is used at MQTT v5.0-spec.

Reply:
But our mqtt version is 3.x now, [http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718063](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718063) 
> The payload of a SUBSCRIBE packet MUST contain at least one Topic Filter / QoS pair. A SUBSCRIBE packet with no payload is a protocol violation